src/ml_tooling/baseclass.py

```diff
@@ -582,7 +504,7 @@ def gridsearch(
                 parameters=parameters,
             )
             for parameters, (train, test) in product(
-                param_grid, cv.split(train_x, train_y, None)
+                param_grid, cv.split(data.train_x, data.train_y, None)
```

Comment:
Maybe the generation of splits should be de handled by a method in dataset. That way we can easliy define it for times series and grouped observations.

Reply:
Agreed, but think that's a more advanced feature we would have to think about how to implement - Make this an issue?

Reply:
Sure :-) I will create an issue.